logback-core/src/main/java/ch/qos/logback/core/net/ssl/SSLConfigurableSocket.java

```diff
@@ -60,4 +64,28 @@ public void setWantClientAuth(boolean state) {
         delegate.setWantClientAuth(state);
     }
 
+    public void setDisableHostnameVerification(boolean disableHostnameVerification) {
+        if (disableHostnameVerification) {
+            return;
+        }
+        SSLParameters sslParameters = delegate.getSSLParameters();
+        try {
+            /*
+             * SSLParameters.setEndpointIdentificationAlgorithm(...)
+             * is only available from Java 7, so we use reflection
+             * to check whether it is available.
+             */
+            Method setEndpointIdentificationAlgorithmMethod = SSLParameters.class.getMethod("setEndpointIdentificationAlgorithm", String.class);
+            setEndpointIdentificationAlgorithmMethod.invoke(sslParameters, "HTTPS");
+            delegate.setSSLParameters(sslParameters);
+        } catch (NoSuchMethodException e) {
+            // We don't do anything when the method is not available.
```

Comment:
Is that really a good idea? I don't set disableHostnameVerification. But if I don't have that method, the hostname verification silently fails. I personally would prefer another RuntimeException. Or at least a warning printed. In the current situation it is impossible for me as a user to know if hostname verification is actually enabled or if it silently failed.
If it throws an exception because it is not supported, then the hostname verification can be still explicitly disabled.